src/plugins/mer/mervirtualmachinesettingswidget.cpp

```diff
@@ -112,22 +130,27 @@ void MerVirtualMachineSettingsWidget::initGui()
     });
 
     ui->memoryInfoLabel->setPixmap(Utils::Icons::INFO.pixmap());
-    ui->memoryInfoLabel->setToolTip(
-            QLatin1String("<font color=\"red\">")
-            + tr("Stop the virtual machine to unlock this field for editing.")
-            + QLatin1String("</font>"));
-
     ui->cpuInfoLabel->setPixmap(Utils::Icons::INFO.pixmap());
-    ui->cpuInfoLabel->setToolTip(
-            QLatin1String("<font color=\"red\">")
-            + tr("Stop the virtual machine to unlock this field for editing.")
-            + QLatin1String("</font>"));
-
     ui->storageSizeInfoLabel->setPixmap(Utils::Icons::INFO.pixmap());
-    ui->storageSizeInfoLabel->setToolTip(
-            QLatin1String("<font color=\"red\">")
-            + tr("Stop the virtual machine to unlock this field for editing.")
-            + QLatin1String("</font>"));
+}
+
+void MerVirtualMachineSettingsWidget::setToolTip(QLabel *label, const QString &toolTipText)
+{
+   label->setToolTip(QLatin1String("<font color=\"red\">")
+                     + toolTipText
+                     + QLatin1String("</font>"));
+}
+
+void MerVirtualMachineSettingsWidget::setStorageSizeLimits()
+{
+    // Prohibit adding more than MAX_STORAGE_SIZE_INCREMENT_GB in one step, because there is no way to go back
+    // with VirtualBox
+    const double currentValue = ui->storageSizeGbSpinBox->value();
+    const double maximumSizeGb = (m_features & VirtualMachine::GrowStorageSize)
+        ? currentValue + MAX_STORAGE_SIZE_INCREMENT_GB : currentValue;
+    const double minimumSizeGb = (m_features & VirtualMachine::ShrinkStorageSize)
+        ? 0 : currentValue;
```

Comment:
great but the ugly extra parentheses :) I would prefer wrapping each operand of `?:` onto a separate line

```
    const double maximumSizeGb = m_features & VirtualMachine::GrowStorageSize
        ? currentValue + MAX_STORAGE_SIZE_INCREMENT_GB
        : currentValue;
    const double minimumSizeGb = m_features & VirtualMachine::ShrinkStorageSize
        ? 0
        : currentValue;
```

Reply:
Just following Qt Creator coding convention with the parenthesis: https://doc-snapshots.qt.io/qtcreator-extending/coding-style.html#parentheses

Reply:
That's a bit different case :) check `grep -e '^[[:space:]]*?' -C1 -r src/plugins/` for reference. Slightly related is also what I pasted before under some review https://pastebin.com/raw/PRWJsWip

Reply:
Ok, let's then go against the coding convention. Sure, they've done it themselves in quite a few places so I guess it's understandable. Or maybe they have updated the convention but haven't updated the code everywhere yet?

Reply:
Done. I still want to emphasize that:
1) I prefer using the parenthesis
2) So does the official Qt Creator coding style.
But as this bug fix needs to go in, I'm not going to waste time arguing about this any further.

Reply:
My understanding is that the coding convention deals with two issues: code readability and error prevention. When comes to ternary operator use, the former is more conveniently solved by line wrapping and the latter is not likely to be relevant given that ternary operator has as low precedence as the assignment operators have. Of course the situation would be different at the level of its operands, so e.g.

```
return (A && B) || C
    ? (a + b) & c
    : 0;
```

would be prefered.